### PR TITLE
test: refactor tests, expand with `fuzz` + `fork` coverage

### PR DESCRIPTION
**Pull request by igorroncevic**
## Type of Change

- [ ] New Feature
- [ ] Bug Fix
- [x] Refactor
- [x] Other

## Description

- Expanded the automated test suite from **210** to **955** checks (**+745** tests/invariants).
- Refactored tests into a consistent structure and style across `unit`, `integration`, `fork`, and `invariant` suites:
  - Unified naming convention: `test_{unit|integration|fuzz|invariant|fork}_{functionName}_{success|revertsWith}_...`
  - Consistent section banners and comment blocks for requirements/effects/interactions intent
  - Split and reorganized suites into function-focused files with shared base test contracts
- Added substantial fuzz coverage to exercise edge cases and input-space combinations, with **247 fuzz tests** currently in the suite
- Added substantial real-network validation coverage with fork tests, with **89 fork tests** across **7 suites** 
- Improved overall coverage further by closing gaps on core protocol paths, including success/revert and event/state transitions.
- Increased maintainability and reviewability of tests by centralizing shared harnesses/mocks/helpers and reducing duplicated setup logic.

## Critical Parts

- Naming consistency and test organization conventions across newly split suites
- Fuzz test assumptions/bounds and whether they model realistic constraints
- Fork test setup assumptions (pinned blocks, chain-specific addresses, and on-chain integration expectations)
- High-risk protocol paths now covered by additional tests:
  - liquidation and borrower position flows
  - bridge/rebalancer message paths and authorization checks
  - operator and pause/role enforcement behavior

## How Has This Been Tested?

- Local/CI validation commands:
  - `make fmt`
  - `make lint`
  - `make test`
- Branch-level suite growth verification:
  - `test_* + invariant_*` checks at `HEAD`: `955`
  - Fuzz-named tests at `HEAD`: `247`
  - Fork tests at `HEAD`: `89`

## Checklist

- [x] Added code follows Solidity best practices and style guidelines similar to the rest of the project.
- [x] Added code has been reviewed for potential security vulnerabilities.
- [x] Added code has been tested thoroughly.
- [x] All tests pass successfully when running `make test`.
- [x] NatSpec documentation is up to date and accurate.
- [x] NatSpec for external dependencies is added to the code, with file's appropriate Git blob linked (e.g. `https://github.com/malda-protocol/malda-lending/blob/6da3402491b0597592e7ce77df035b59e43e1167/src/mToken/host/mErc20Host.sol`).

## Additional Context

- https://linear.app/malda/issue/MAL-1559/3-full-revamp-of-tests-base-contracts-more-test-cases-fork-tests-fuzz

### CODE REVIEW

File: test/invariant/mToken.invariant.t.sol [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        uint256 balance = mToken.balanceOf(actor);
+        amount = bound(amount, 1, balance);
+        vm.startPrank(actor);
+        mToken.redeem(amount);
+        vm.stopPrank();
+        ghostTotalSupply -= amount;
+    }
+}
+
+contract MTokenInvariantTest is BaseMTokenTest {
```
Comment (0xnimrodel) [inline test/invariant/mToken.invariant.t.sol]: `setUp` never calls `operator.supportMarket()`, so every mint/redeem silently reverts and the assertion passes by default with both sides at 0.

Image

**Review by 0xnimrodel — COMMENTED** [review]
The inline comments cover the key findings.

**Review by 0xnimrodel — APPROVED** [review]

**Review by Barnadrot — APPROVED** [review]
lgtm
